tfjs-backend-webgpu/src/kernels/Dilation2D.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, Dilation2D, Dilation2DAttrs, Dilation2DInputs, KernelConfig, KernelFunc, TensorInfo} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {Dilation2DProgram} from '../dilation_webgpu';
+import {reshape} from './Reshape';
+
+export function dilation2D(args: {
+  inputs: Dilation2DInputs,
+  attrs: Dilation2DAttrs,
+  backend: WebGPUBackend
+}): TensorInfo {
+  const {inputs, backend, attrs} = args;
+  const {x, filter} = inputs;
+  const {strides, pad, dilations} = attrs;
+
+  const convInfo = backend_util.computeDilation2DInfo(
+      x.shape as [number, number, number, number],
+      filter.shape as [number, number, number], strides, pad,
+      'NHWC' /* dataFormat */, dilations);
+  const padInfo = [convInfo.padInfo.top, convInfo.padInfo.left];
+  const uniformData = [
+    {type: 'int32', data: [convInfo.filterHeight, convInfo.filterWidth]},
+    {type: 'int32', data: [...padInfo]},
+    {type: 'int32', data: [convInfo.strideHeight, convInfo.strideWidth]},
+    {type: 'int32', data: [convInfo.dilationHeight, convInfo.dilationWidth]}
+  ];
+
+  const program = new Dilation2DProgram(convInfo);
+  const out =
+      backend.runWebGPUProgram(program, [x, filter], 'float32', uniformData);
+
+  const outReshaped =
+      reshape({inputs: {x: out}, backend, attrs: {shape: convInfo.outShape}});
```

Comment:
Sorry, forgot one. Why you still need the additional `reshape` since the `out` shape is already `convInfo.outShape`?

Reply:
Done, I had remove this reshape code. And changed the type of output tensor to rely on `x` dtype.